pkg/controller/network_policy.go

```diff
@@ -558,13 +647,74 @@ func (c *Controller) fetchPolicySelectedAddresses(namespace, protocol string, np
 			for _, podIP := range pod.Status.PodIPs {
 				if podIP.IP != "" && util.CheckProtocol(podIP.IP) == protocol {
 					selectedAddresses = append(selectedAddresses, podIP.IP)
+
+					svcs, err := c.servicesLister.Services(pod.Namespace).List(labels.Everything())
+					if err != nil {
+						klog.Errorf("failed to list svc, %v", err)
+						return nil, nil, fmt.Errorf("failed to list svc, %v", err)
+					}
+					if len(svcs) == 0 {
+						continue
+					}
+					klog.Infof("svc is %v", svcs)
+					svcIPs, err := svcMatchPods(svcs, pod, protocol)
+					if err != nil {
+						return nil, nil, err
+					}
+					klog.Infof("svcIPs is %v", svcIPs)
+					if len(svcIPs) == 0 {
+						continue
+					}
+					selectedAddresses = append(selectedAddresses, svcIPs...)
 				}
 			}
 		}
 	}
+
 	return selectedAddresses, nil, nil
 }
 
+func svcMatchPods(svcs []*corev1.Service, pod *corev1.Pod, protocol string) ([]string, error) {
+	matchSvcs := []string{}
+	// find  svc ip by pod's info
+	for _, svc := range svcs {
+		isMatch, err := isSvcMatchPod(svc, pod)
+		if err != nil {
+			return nil, err
+		}
+		if isMatch {
+			protocolClusterIPs := getProtocolSvcIp(svc.Spec.ClusterIPs, protocol)
+			if len(protocolClusterIPs) != 0 {
+				matchSvcs = append(matchSvcs, protocolClusterIPs...)
+			}
```

Comment:
1. Dual stack feature for serivce was introduced in Kubernetes v1.20, so in earlier versions value of the field `spec.cluserIPs` is always empty. In those cases we should use `spec.clusterIP` instead.
2. If a service has no cluster IP, value of `spec.clusterIPs` will be `[]string{"None"}`.

 You can refer #1146.